x-pack/plugins/security_solution/public/flyout/document_details/right/footer.tsx

```diff
@@ -59,19 +99,143 @@ export const PanelFooter: FC<PanelFooterProps> = ({ isPreview }) => {
+            { category: curr.category, field: curr.field },
+            dataFormattedForFieldBrowser
+          ),
+        }),
+        {} as AlertSummaryData
+      ),
+    [dataFormattedForFieldBrowser]
+  );
+  const {
+    exceptionFlyoutType,
+    openAddExceptionFlyout,
+    onAddExceptionTypeClick,
+    onAddExceptionCancel,
+    onAddExceptionConfirm,
+  } = useExceptionFlyout({
+    refetch: refetchAll,
+    isActiveTimelines: isActiveTimeline(scopeId),
+  });
+
+  // event filter interaction
+  const { closeAddEventFilterModal, isAddEventFilterModalOpen, onAddEventFilterClick } =
+    useEventFilterModal();
+
+  // osquery interaction
+  const [isOsqueryFlyoutOpenWithAgentId, setOsqueryFlyoutOpenWithAgentId] = useState<null | string>(
+    null
+  );
+  const closeOsqueryFlyout = useCallback(() => {
+    setOsqueryFlyoutOpenWithAgentId(null);
+  }, [setOsqueryFlyoutOpenWithAgentId]);
+  const alertId = useMemo(
+    () => (dataAsNestedObject?.kibana?.alert ? dataAsNestedObject?._id : null),
+    [dataAsNestedObject?._id, dataAsNestedObject?.kibana?.alert]
+  );
+
+  if (isPreview) return null;
+
+  return (
+    <>
+      <StyledEuiFlyoutFooter data-test-subj={FLYOUT_FOOTER_TEST_ID}>
+        <EuiFlexGroup justifyContent="flexEnd">
+          <EuiFlexItem grow={false}>
+            {dataAsNestedObject && (
+              <TakeActionDropdown
+                dataFormattedForFieldBrowser={dataFormattedForFieldBrowser}
+                dataAsNestedObject={dataAsNestedObject}
+                handleOnEventClosed={closeFlyout}
+                isHostIsolationPanelOpen={isHostIsolationPanelOpen}
+                onAddEventFilterClick={onAddEventFilterClick}
+                onAddExceptionTypeClick={onAddExceptionTypeClick}
+                onAddIsolationStatusClick={showHostIsolationPanelCallback}
+                refetchFlyoutData={refetchFlyoutData}
+                refetch={refetchAll}
+                scopeId={scopeId}
+                onOsqueryClick={setOsqueryFlyoutOpenWithAgentId}
+              />
+            )}
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </StyledEuiFlyoutFooter>
+
+      {openAddExceptionFlyout &&
+        alertSummaryData.ruleId != null &&
+        alertSummaryData.ruleRuleId != null &&
+        alertSummaryData.eventId != null && (
+          <AddExceptionFlyoutWrapper
+            {...alertSummaryData}
+            ruleIndices={ruleIndex}
+            ruleDataViewId={ruleDataViewId}
+            exceptionListType={exceptionFlyoutType}
+            onCancel={onAddExceptionCancel}
+            onConfirm={onAddExceptionConfirm}
+          />
+        )}
+
+      {isAddEventFilterModalOpen && dataAsNestedObject != null && (
+        <EventFiltersFlyout
+          data={dataAsNestedObject}
+          onCancel={closeAddEventFilterModal}
+          // EUI TODO: This z-index override of EuiOverlayMask is a workaround, and ideally should be resolved with a cleaner UI/UX flow long-term
```

Comment:
is there an EUI ticket for improving the z-index?

Reply:
yeah I vaguely remember talking with EUI months ago when I started adding these. I remember chatting with @cee-chen  at the time and thought they had a ticket, but a quick search isn't giving me back anything...

Reply:
I do not remember chatting about this, sorry! The closest issue is probably https://github.com/elastic/eui/issues/7443, although it specifically wants to stack push/overlay flyouts rather than multiple overlay flyouts (not sure if that's what's happening here?).

@MichaelMarcialis do you mind chiming in here about the advisability of multiple/stacking flyouts? IIRC last we discussed this, we were generally leaning towards avoiding it.

Reply:
Thanks for the ping, @cee-chen. Yes, generally speaking, I would suggest avoiding scenarios where we're attempting to stack flyouts (be they overlay-only flyouts or an overlay on top of a push flyout). If we find ourselves approaching such a situation, it's likely a sign that the UX we're trying to craft has outgrown what we'd consider the standard flyout experience. At that point, I would argue that are likely better alternatives to consider, such as:

- Moving the content from the originating flyout to it's own separate page/view.
- Altering the IA of the originating flyout so that the user is taken one level/layer deeper into that single flyout component (rather than opening an entirely new flyout on top). An example of this can be seen in Lens layer configuration flyouts.

TL;DR: Consider exploring alternatives such as the above or others before resigning to a stacked flyout experience.